### PR TITLE
[Dist] Add support for Parquet-formatted edges files, remove some assumptions on edge file number.

### PR DESCRIPTION
**Pull request by thvasilo**
## Description

This PR allows the edges files to be loaded from Parquet files apart from CSV.

I haven't added any unit tests specific to this change, if needed let me know.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by thvasilo** [comment]
HI @Rhett-Ying I see some lint failures in the code I changed as well as pre-existing code in files I touch in this PR. How do you suggest we resolve this? 

Is there a specific linter  (like `black`) that  will also apply the changes so I can update the PR?

**Comment by Rhett-Ying** [comment]
Hi @thvasilo It's ok to ignore the lint check in this PR. All the code under `//tools/` are not formatted yet and It's an exception. we'll formatting them in the near future.

File: tools/distpartitioning/dataset_utils.py [hunk 1]
```diff
@@ -528,16 +527,24 @@ def get_dataset(input_dir, graph_name, rank, world_size, num_parts, schema_map):
...
+                parse_options=pyarrow.csv.ParseOptions(delimiter=' ')
+                with pyarrow.csv.open_csv(edge_file, read_options=read_options, parse_options=parse_options) as reader:
+                    for next_chunk in reader:
+                        if next_chunk is None:
+                            break
+
+                        next_table = pyarrow.Table.from_batches([next_chunk])
+                        src_ids.append(next_table['f0'].to_numpy())
+                        dst_ids.append(next_table['f1'].to_numpy())
+            elif etype_info[constants.STR_FORMAT][constants.STR_NAME] == constants.STR_PARQUET:
```
Comment (Rhett-Ying) [inline tools/distpartitioning/dataset_utils.py:541]: we'd better add test to cover this scenario. see more details in `//tests/tools/test_dist_part.py` and `//tests/tools/utils.py`.

Comment (Rhett-Ying) [reply tools/distpartitioning/dataset_utils.py:541]: https://github.com/dmlc/dgl/blob/00f2099934ca78a3c156c4222edcc3245cee4e1c/tests/tools/utils.py#L134-L151

**Comment by Rhett-Ying** [comment]
`cd ~/workspace/dgl_0`

`PYTHONPATH=tools:tools/distpartitioning:tests:$PYTHONPATH python3 -m  pytest tests/tools/test_dist_part.py::test_pipeline_basics`

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@thvasilo CI failed, pls check the error details in report path shown above.

**Comment by Rhett-Ying** [comment]
Pls try with the command I shared above to run tests in your local.

**Comment by thvasilo** [comment]
Thanks @Rhett-Ying I was able to test locally with the above test, amended my last commit to fix the error. Let's see how the CI goes.

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Comment by thvasilo** [comment]
@Rhett-Ying can you review this again?

**Comment by Rhett-Ying** [comment]
@dgl-bot

**Review by Rhett-Ying — APPROVED** [review]
LGTM

**Comment by Rhett-Ying** [comment]
@thvasilo As `//tests/tools/test_parmetis.py::test_parmetis_preprocessing` is skipped in CI, pls make sure the PR works as expected on your local.

**Pull request merged**